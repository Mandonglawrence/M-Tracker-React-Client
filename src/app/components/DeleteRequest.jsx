import React from 'react';
import PropTypes from 'prop-types';
import '../assets/css/createRequestForm.css';

const DeleteRequest = ({ handleClick }) => (
  <div
    className="modal fade"
    id="deleteRequest"
    tabIndex="-1"
    role="dialog"
    aria-labelledby="deleteRequest"
    aria-hidden="true"
  >
    <div className="modal-dialog modal-dialog-centered" role="document">
      <div className="modal-content">
        <div className="modal-header p-2 w3-flat-midnight-blue">
          <h5 className="modal-title" id="exampleModalCenterTitle">
            Delete Request?
          </h5>
          <button
            type="button"
            id="closeDeleteModal"
            className="close p-4"
            data-dismiss="modal"
            aria-label="Close"
          >
            <span aria-hidden="true">
              &times;
            </span>
          </button>
        </div>
        <div className="modal-body">
          <div>
            <p>
              This action can not be undone, are you sure you want to delete this request?
            </p>
            <button
              type="button"
              id="closeDeleteModal"
              className="btn btn-danger"
              data-dismiss="modal"
              aria-label="Close"
            >
              Cancel
            </button>
            <button
              type="button"
              id="closeDeleteModalTrue"
              className="btn btn-success ml-4"
              data-dismiss="modal"
              aria-label="Close"
              onClick={handleClick}
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
);

DeleteRequest.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

export default DeleteRequest;
