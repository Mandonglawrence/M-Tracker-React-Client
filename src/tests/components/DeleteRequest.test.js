import React from 'react';
import { shallow } from 'enzyme';
import DeleteRequest from '../../app/components/DeleteRequest';

const props = {
  handleClick: jest.fn(),
};

describe('Delete request component', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<DeleteRequest {...props} />);
    expect(wrapper).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
