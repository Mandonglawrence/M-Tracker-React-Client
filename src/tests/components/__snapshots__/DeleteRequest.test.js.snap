// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete request component should render without errors 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="deleteRequest"
  className="modal fade"
  id="deleteRequest"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog modal-dialog-centered"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header p-2 w3-flat-midnight-blue"
      >
        <h5
          className="modal-title"
          id="exampleModalCenterTitle"
        >
          Delete Request?
        </h5>
        <button
          aria-label="Close"
          className="close p-4"
          data-dismiss="modal"
          id="closeDeleteModal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <div>
          <p>
            This action can not be undone, are you sure you want to delete this request?
          </p>
          <button
            aria-label="Close"
            className="btn btn-danger"
            data-dismiss="modal"
            id="closeDeleteModal"
            type="button"
          >
            Cancel
          </button>
          <button
            aria-label="Close"
            className="btn btn-success ml-4"
            data-dismiss="modal"
            id="closeDeleteModalTrue"
            onClick={[MockFunction]}
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
